src/Ucommerce.Avenue.Umbraco/Controllers/SearchController.cs

```diff
@@ -1,64 +1,63 @@
-                products = Product.Find(p =>
-                    p.VariantSku == null
-                    && p.DisplayOnSite
-                    &&
-                    (
-                        p.Sku.Contains(keyword)
-                        || p.Name.Contains(keyword)
-                        || p.ProductDescriptions.Any(d => d.DisplayName.Contains(keyword)
-                                                          || d.ShortDescription.Contains(keyword)
-                                                          || d.LongDescription.Contains(keyword)
-                        )
-                    )
-                );
+                products = ProductIndex.Find()
+                    .Where(p => p.VariantSku == null &&
+                                (
+                                    p.Sku.Contains(keyword)
+                                    || p.Name.Contains(keyword)
+                                    || p.DisplayName.Contains(keyword)
+                                    || p.ShortDescription.Contains(keyword)
+                                    || p.LongDescription.Contains(keyword)
+                                ))
+                    .ToList();
             }
 
-            foreach (var product in products.Where(x => x.DisplayOnSite))
+            var productPriceCalculationResult = CatalogLibrary.CalculatePrices(products.Select(p => p.Guid).ToList());
+            var pricesPerProductId = productPriceCalculationResult.Items.ToLookup(item => item.ProductGuid);
+
+            foreach (var product in products)
             {
-                productsViewModel.Products.Add(new ProductViewModel()
+                var productPriceCalculationResultItem = pricesPerProductId[product.Guid].First();
+                productsViewModel.Products.Add(new ProductViewModel
                 {
-                    // Url = UrlService.GetUrl(CatalogContext.CurrentCatalog, new[] {product}),
-                    Name = product.DisplayName(),
+                    Url = UrlService.GetUrl(CatalogContext.CurrentCatalog, new[] {product}),
+                    Name = product.DisplayName,
                     Sku = product.Sku,
-                    IsVariant = product.IsVariant,
-                    LongDescription = product.LongDescription(),
-                    // PriceCalculation = CatalogLibrary.CalculatePrices(new List<Guid>{product.Guid}),
-                    ThumbnailImageUrl = ObjectFactory.Instance.Resolve<IImageService>()
-                        .GetImage(product.ThumbnailImageMediaId).Url,
+                    IsVariant = !string.IsNullOrWhiteSpace(product.VariantSku),
+                    LongDescription = product.LongDescription,
+                    PriceCalculation = new ProductPriceCalculationViewModel
+                    {
+                        YourPrice = productPriceCalculationResultItem.PriceInclTax.ToString("C"),
+                        ListPrice = productPriceCalculationResultItem.ListPriceInclTax.ToString("C")
```

Comment:
What is the reason of passing string `C`?

Reply:
so the decimal is displayed 'pretty'. With this it's like this: `$320.85`, without it something like this: `320.85000`

Reply:
Yes, but we need to redo this part. There's a bunch of specialized code for formatting money. ToString is too simplistic (it doesn't look up the currency symbol in based on the pricegroup for instance).

Reply:
yes of course, but I think we should tackle that after the workshop

Reply:
:+1: